packages/admin/cms-admin/src/pages/pageTree/useCopyPastePages/sendPages.ts

```diff
@@ -0,0 +1,391 @@
+                        ?.replaceWithId,
+                    name,
+                    slug,
+                    hideInMenu: node.hideInMenu,
+                    attachedDocument: {
+                        type: node.documentType,
+                    },
+                    parentId: newParentId,
+                    pos: options.targetPos ? options.targetPos + posOffset : undefined,
+                },
+                contentScope: scope,
+                category: node.category,
+            },
+            context: LocalErrorScopeApolloContext,
+        });
+        if (!data?.createPageTreeNode.id) {
+            throw Error("Did not receive new uuid for page tree node");
+        }
+
+        // 1c. Copy all files used on page to target scope
+        const fileIdsToCopyDirectly: string[] = [];
+        if (node?.document != null) {
+            for (const damFile of fileDependenciesFromDocument(documentType, node.document)) {
+                if (dependencyReplacements.some((replacement) => replacement.type == "DamFile" && replacement.originalId === damFile.id)) {
+                    //already copied
+                } else if (damFile.fileUrl.startsWith(blockContext.damConfig.apiUrl)) {
+                    //our own api, no need to download&upload
+                    if (!hasDamScope || isEqual(damFile.scope, damScope)) {
+                        //same scope, same server, no need to copy
+                    } else {
+                        //batch copy below
+                        fileIdsToCopyDirectly.push(damFile.id);
+                    }
+                } else {
+                    if (damFile.fileUrl.match(/^https?:\/\/(localhost|.*\.dev\.vivid-planet\.cloud|192\.168\.\d{1,3}\.\d{1,3}):\d{2,4}/)) {
+                        //source is local dev server, download client side and upload
+                        const fileResponse = await fetch(damFile.fileUrl);
+                        const fileBlob = await fileResponse.blob();
+                        const file = new File([fileBlob], damFile.name, { type: damFile.mimetype });
+                        const formData = new FormData();
+                        formData.append("file", file);
+                        if (hasDamScope) formData.append("scope", JSON.stringify(damScope));
+                        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                        formData.append("folderId", inboxFolderIdForCopiedFiles!);
+                        if (damFile.title) formData.append("title", damFile.title);
+                        if (damFile.altText) formData.append("altText", damFile.altText);
+                        if (damFile.license) formData.append("license", JSON.stringify(damFile.license));
+                        if (damFile.image?.cropArea) formData.append("imageCropArea", JSON.stringify(damFile.image.cropArea));
+                        formData.append("copyOfId", damFile.id);
```

Comment:
In the FilesService there is the following code:

```
let copyOf: FileInterface | null = null;
if (copyOfId) {
    copyOf = await this.findOneById(copyOfId);
    if (!copyOf) throw new Error("Copy of file not found");
}
```

Meaning if the `copyOfId` doesn't exist in the current instance, the upload fails. So if you download from another instance, `copyOfId` should be `undefined` otherwise, I guess, the upload will fail here (I didn't test this).

Reply:
yes, this could possibly fail. Would you really always remove it?

Reply:
and with the new implementation it can easily be faked. Because we don't really do anything with it so far, it is no security issue (so far)

possible fix: remove it completely for upload and download

Reply:
> possible fix: remove it completely for upload and download

I think that's okay. If we download from another instance, we never have the file id in our instance

Reply:
that is the plan

Reply:
Removed copyOf from the upload/import apis